torch/multiprocessing/spawn.py

```diff
@@ -17,6 +17,9 @@ def __init__(self, msg: str, error_index: int, pid: int):
         self.error_index = error_index
         self.pid = pid
 
+    def __reduce__(self):
+        return (self.__class__, self.args + (self.error_index, self.pid) + {})
+
```

Comment:
You can do the following to achieve such behavior. Using `args` is too hacky for me.
```py
    def __reduce__(self):
        return type(self), (self.msg, self.error_index, self.pid)
```

Reply:
There is no `msg` attribute, but I did refactor to make it more readable.

Reply:
What you are doing now is assuming there is key value `msg` in `self.args`. I prefer attaching `msg` to each Error instance.